build/scripts/scaffold-integration-test.groovy

```diff
@@ -1,131 +0,0 @@
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-import java.util.stream.Collectors
-import java.nio.charset.Charset
-import java.nio.charset.StandardCharsets
-import java.nio.file.Files
-import java.nio.file.Path
-import java.nio.file.Paths
-
-import freemarker.cache.ClassTemplateLoader
-import freemarker.cache.FileTemplateLoader
-import freemarker.cache.MultiTemplateLoader
-import freemarker.cache.TemplateLoader
-import freemarker.template.Configuration
-import freemarker.template.Template
-import freemarker.template.TemplateException
-import freemarker.template.TemplateExceptionHandler
-
-import io.quarkus.maven.CreateExtensionMojo.TemplateParams
-import io.quarkus.maven.utilities.PomTransformer
-import io.quarkus.maven.utilities.PomTransformer.Transformation
-
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
-
-/* Keep in sync with the current file name */
-@groovy.transform.Field static final String currentScript = "scaffold-integration-test.groovy"
-@groovy.transform.Field static final Logger log = LoggerFactory.getLogger('scaffold-integration-test')
-final Path extensionsDir = project.basedir.toPath()
-final Path templatesDir = extensionsDir.resolve('../build/create-extension-templates')
-final String artifactIdBase = properties['quarkus.artifactIdBase']
-final Path itestDir = extensionsDir.resolve('../integration-tests/' + artifactIdBase)
-final Charset charset = StandardCharsets.UTF_8
-
-final nameBase = properties['quarkus.nameBase'] == null ? artifactIdBase.split(' ').collect{it.capitalize()}.join(' ') : artifactIdBase
-
-Files.createDirectories(itestDir)
-
-final Configuration cfg = new Configuration(Configuration.VERSION_2_3_28)
-cfg.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER)
-cfg.setTemplateLoader(createTemplateLoader(templatesDir))
-cfg.setDefaultEncoding('UTF-8')
-cfg.setInterpolationSyntax(Configuration.SQUARE_BRACKET_INTERPOLATION_SYNTAX)
-cfg.setTagSyntax(Configuration.SQUARE_BRACKET_TAG_SYNTAX)
-
-TemplateParams model = new TemplateParams()
-
-model.artifactIdBase = artifactIdBase
-model.artifactIdBaseCamelCase = toCapCamelCase(model.artifactIdBase)
-model.version = project.version
-
-model.nameBase = nameBase
-
-model.javaPackageBase = getJavaPackage('org.apache.camel.quarkus', 'component', artifactIdBase)
-final String javaPackageBasePath = model.javaPackageBase.replace('.', '/')
-
-evalTemplate(cfg, "integration-test-pom.xml", itestDir.resolve('pom.xml'), charset, model)
-evalTemplate(cfg, "TestResource.java", itestDir.resolve('src/main/java/' + javaPackageBasePath + '/it/'+ model.artifactIdBaseCamelCase + 'Resource.java'), charset, model)
-evalTemplate(cfg, "TestRouteBuilder.java", itestDir.resolve('src/main/java/' + javaPackageBasePath + '/it/'+ model.artifactIdBaseCamelCase + 'RouteBuilder.java'), charset, model)
```

Comment:
TestRouteBuilder.java is not used by the quarkus-maven-plugin mojo because it is Camel specific. So either we should keep a simplified scaffold-integration-test.groovy to keep RouteBuilders is generated or we should remove TestRouteBuilder.java from build/create-extension-templates.

I am undecided whether having a generated RouteBuilder stub in the itest module is enough for keeping scaffold-integration-test.groovy.

Reply:
I'm for removing it

Reply:
removed it, if you do not like, I can bring it back